asserts/database.go

```diff
@@ -461,3 +479,45 @@ var DefaultCheckers = []Checker{
 	CheckTimestampVsSigningKeyValidity,
 	CheckCrossConsistency,
 }
+
+// no-authority assertion checkers
+
+// NoAuthorityCheckSignature checks that the signature is valid.
+func NoAuthorityCheckSignature(assert Assertion, roDB RODatabase, checkTime time.Time) error {
+	selfSigned, ok := assert.(selfSignedAssertion)
+	if !ok {
+		panic(fmt.Errorf("cannot check non-self-signed assertion type %q", assert.Type().Name))
+	}
+	signingKey := selfSigned.signKey()
+	content, encSig := assert.Signature()
+	signature, err := decodeSignature(encSig)
+	if err != nil {
+		return err
+	}
+	err = signingKey.verify(content, signature)
+	if err != nil {
+		return fmt.Errorf("failed signature verification: %v", err)
+	}
+	return nil
+}
+
+// A noAuthorityConsistencyChecker performs further checks based on the full
+// assertion database knowledge and its own signing key.
+type noAuthorityConsistencyChecker interface {
+	noAuthorityCheckConsistency(roDB RODatabase) error
```

Comment:
I don't think that makes sense in this case: `checkConsistency` (which this is modelled on) doesn't have it.